### ISSUE TITLE
Request: Intellisense ordering by context relevance

### ISSUE THREAD

**Issue by DracheTek** [microsoft/vscode-cpptools#11738]
### Feature Request

When typing parameter for functions, I am hoping that Intellisens put the most relevant entries on top.
![微信截图_20231125102830](https://github.com/microsoft/vscode-cpptools/assets/11999260/a8ffd933-6dc4-4377-9d33-a0d26a9530c2)

take this for example, function prototype is follows:
```c
void PORT_PinmuxConfig(PORT_ID_t portId, PORT_GPIONO_t gpioNo,PORT_PinMuxFunc_t pinMuxFunc)
```
I defined enum type for this parameter elsewhere, and I'm hoping the enum entry {PORT_A,PORT_B,PORT_C......} come on top of the list, instead of all keywords coming out together in alphabetic order
Visual Studio support this feature natively IIRC.

**Comment by sean-mcmanus** [comment]
Yes, we're aware of this feature -- we had a pre-existing issue tracking this, but I can't seem to locate it right now.

**Comment by bobbrow** [comment]
I could not find the existing ticket for this, so let's use this one instead.
